payment-manager/src/main/java/com/qpay/paymentmanager/service/impl/PaymentServiceImpl.java

```diff
@@ -23,6 +30,10 @@ public class PaymentServiceImpl implements PaymentService {
 
     private final WalletService walletService;
 
+    private final WebClientConfig webClientConfig;
+
+    private final ExecutorService executorService = Executors.newFixedThreadPool(5);
```

Comment:
why 5 ?
Maybe it's better to use newWorkStealingPool.

Also you need to extract it to a config file and create it as a bean